libgit2-sys/build.rs

```diff
@@ -1,32 +1,59 @@
 use std::env;
+use std::ffi::OsString;
 use std::fs;
 use std::io;
 use std::path::{Path, PathBuf};
 use std::process::Command;
 
+fn env_var(name: &str) -> Option<OsString> {
+    let var = env::var_os(name);
+    println!("cargo:rerun-if-env-changed={}", name);
+    var
+}
+
+fn use_system_libgit2() -> Result<(), ()> {
+    let mut cfg = pkg_config::Config::new();
+    if let Ok(lib) = cfg.range_version("1.6.4".."1.7.0").probe("libgit2") {
+        for include in &lib.include_paths {
+            println!("cargo:root={}", include.display());
+        }
+        Ok(())
+    } else {
+        Err(())
+    }
+}
+
 fn main() {
-    let https = env::var("CARGO_FEATURE_HTTPS").is_ok();
-    let ssh = env::var("CARGO_FEATURE_SSH").is_ok();
-    let vendored = env::var("CARGO_FEATURE_VENDORED").is_ok();
-    let zlib_ng_compat = env::var("CARGO_FEATURE_ZLIB_NG_COMPAT").is_ok();
+    let https = env_var("CARGO_FEATURE_HTTPS").is_some();
+    let ssh = env_var("CARGO_FEATURE_SSH").is_some();
+    let vendored = env_var("CARGO_FEATURE_VENDORED").is_some();
+    let zlib_ng_compat = env_var("CARGO_FEATURE_ZLIB_NG_COMPAT").is_some();
+    let no_vendor = env_var("LIBGIT2_NO_VENDOR").map_or(false, |s| s != "0");
```

Comment:
We should doc this new environment variable about why and when people should use it. Maybe in `README.md` or elsewhere.